index.js

```diff
@@ -6,6 +6,22 @@ import pRetry from 'p-retry';
 // Matches '/<re/po>/tree/<ref>/<dir>'
 const urlParserRegex = /^[/]([^/]+)[/]([^/]+)[/]tree[/]([^/]+)[/](.*)/;
 
+async function isResponseLfs(response) {
+	const length = Number.parseInt(await response.headers.get('content-length'), 10);
+	if (length > 128 && length < 140) {
+		const text = await response.text();
+		return text.startsWith('version https://git-lfs.github.com/spec/v1');
+	}
+}
+
+async function handleLfs(controller, user, repository, ref, path) {
```

Comment:
You can pass around the signal rather than the controller

```suggestion
async function handleLfs(user, repository, ref, path, signal) {
```